library/cpp/threading/local_executor/tbb_local_executor.cpp

```diff
@@ -0,0 +1,33 @@
+#include "tbb_local_executor.h"
+
+void NPar::TTbbLocalExecutor::SubmitAsyncTasks(TLocallyExecutableFunction exec, int firstId, int lastId) {
+    for (int i = firstId; i < lastId; ++i) {
+        Group.run([=] { exec(i); });
+    }
+}
+
+int NPar::TTbbLocalExecutor::GetThreadCount() const noexcept {
+    return NumberOfTbbThreads;
+}
+
+int NPar::TTbbLocalExecutor::GetWorkerThreadId() const noexcept {
+    return tbb::task_arena::current_thread_index();
+}
+
+void NPar::TTbbLocalExecutor::Exec(TIntrusivePtr<ILocallyExecutable> exec, int id, int flags) {
+    if (flags & WAIT_COMPLETE) {
+        exec->LocalExec(id);
+    } else {
+        SubmitAsyncTasks([=] (int id) { exec->LocalExec(id); }, id, id + 1);
+    }
+}
+
+void NPar::TTbbLocalExecutor::ExecRange(TIntrusivePtr<ILocallyExecutable> exec, int firstId, int lastId, int flags) {
+    if (flags & WAIT_COMPLETE) {
+        tbb::this_task_arena::isolate([&]{
```

Comment:
please capture by value to avoid multiple tsan complaints

Reply:
done. Thanks